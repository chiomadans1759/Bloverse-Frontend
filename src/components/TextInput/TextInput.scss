@import 'sass-rem';

.form-group {
  margin-bottom: rem(8px);
}

.defaultClass {
  height: rem(38px);
  width: rem(358px);
  background-color: rgba(0, 0, 0, 0.45) !important;
  color: #B7B7B7 !important;
  border: solid;
  border-width: rem(1px);
  border-color: rgba(255, 255, 255, 0.15) !important;
  border-radius: 0 !important;
  font-size: rem(16px);
}

.form-control::-webkit-input-placeholder {   /* WebKit, Blink, Edge */
  color: #B7B7B7; 
}

.form-control:-moz-placeholder {  /* Mozilla Firefox 4 to 18 */
  color: #B7B7B7; 
} 

.form-control::-moz-placeholder {   /* Mozilla Firefox 19+ */
  color: #B7B7B7 
}

.form-control:-ms-input-placeholder {  /* Internet Explorer 10-11 */
  color: #B7B7B7; 
} 

.form-control::-ms-input-placeholder { 
  color: #B7B7B7; 
} 

/* Extra small devices (phones, 600px and down) */
@media (min-width: 320px) and (max-width: 480px) { 
  .defaultClass {
    width: 19rem;
  }
}
