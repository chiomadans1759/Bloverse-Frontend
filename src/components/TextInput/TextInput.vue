<template>
  <input
    class="defaultClass"
    :id="id"
    :name="name"
    :value="value"
    :type="type"
    :placeholder="placeholder"
    :class="customClass"
    @change="onChange"
    />
</template>

<script>
export default {
  name: 'TextInput',
  props: {
    id: {
      type: String
    },
    name: {
      type: String,
    },
    value: {
      type: String
    },
    type: {
      type: String
    },
    placeholder: {
      type: String
    },
    customClass: {
      type: String
    },
    onChange: {
      type: Function
    }
  },
}
</script>

<style lang="scss" scoped>
  @import './style.scss';
</style>
