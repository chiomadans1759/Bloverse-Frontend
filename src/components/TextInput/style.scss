@import 'sass-rem';

.defaultClass {
  height: rem(38px);
  width: rem(302px);
  background-color: #ffffff;
  border: solid;
  border-width: rem(1px);
  border-color: rgba(255, 255, 255, 0.15);
  padding-left: rem(16px);
  padding-top: rem(11px);
  padding-right: rem(16px);
  padding-bottom: rem(11px);
  border-radius: 0;
  color: #B7B7B7;
  outline: none;
}
