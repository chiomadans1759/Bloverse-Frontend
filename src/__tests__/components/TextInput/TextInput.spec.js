import { mount } from '@vue/test-utils'
import TextInput from '@/components/TextInput/TextInput.vue';

describe('TextInput Component', () => {
  it('should render correctly', () => {
    const wrapper = mount(TextInput);
    expect(wrapper.element).toMatchSnapshot();
  })
});
