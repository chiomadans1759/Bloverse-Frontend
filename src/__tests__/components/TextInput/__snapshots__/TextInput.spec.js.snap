// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput Component should render correctly 1`] = `
<input
  class="defaultClass"
/>
`;
